src/util/interpolate.ts

```diff
@@ -10,41 +29,71 @@ export const interpolateFactory = (interpolationType: 'number'|'color'|'array'|'
     }
 };
 
-export function number(a: number, b: number, t: number) {
-    return (a * (1 - t)) + (b * t);
+function number(from: number, to: number, t: number): number {
+    return from + t * (to - from);
 }
 
-export function color(from: Color, to: Color, t: number) {
-    return new Color(
-        number(from.r, to.r, t),
-        number(from.g, to.g, t),
-        number(from.b, to.b, t),
-        number(from.a, to.a, t)
-    );
+function color(from: Color, to: Color, t: number, spaceKey: InterpolationColorSpace = 'rgb'): Color {
+    switch (spaceKey) {
+        case 'rgb': {
+            const [r, g, b, alpha] = array(from.rgb, to.rgb, t);
+            return new Color(r, g, b, alpha, false);
+        }
+        case 'hcl': {
```

Comment:
[Optional] Small nit picking: I would consider moving this case to be the last one as `lab` is a shorter method which makes it easier to read.

Reply:
No, in many places where available color spaces are listed they are listed in the order rgb, hcl, lab. I don't see any valid reason to ignore this convention for this function.